### ISSUE TITLE
[🐛 Bug]: Curb Http Client 

### ISSUE THREAD

**Issue by titusfortner** [SeleniumHQ/selenium#14121]
### What happened?

Default Http Client added separate timeout values for open and read timeouts, since that conflicted with the superclass `timeout` setting, we removed that setting (https://github.com/SeleniumHQ/selenium/commit/2965e4e0fa6981ca2fe329ea944c5f3acd0b1caa)

We removed/deprecated the persistent client subclass, but not the curb client subclass. So curb client can no longer set timeouts.

### How can we reproduce the issue?

```shell
Selenium::WebDriver::Remote::Http::Curb.new(timeout: 120)
```

### Relevant log output

```shell
n/a
```

### Operating System

n/a

### Selenium version

Ruby trunk

### What are the browser(s) and version(s) where you see this issue?

n/a

### What are the browser driver(s) and version(s) where you see this issue?

n/a

### Are you using Selenium Grid?

no

**Comment by aguspe** [comment]
Hi @titusfortner I hope you are having a great day, I just created a PR as a bugfix to this #14285 but just a heads up I'm not fully sure if I got the right fix because I might have misunderstood something, so if you have time to review, just let me know what I can fix if something is off, thanks!

**Comment by aguspe** [comment]
Can we consider this resolved due to the merged of #14285 @p0deje ? I'm just looking for something new to pick up and it will help me filter
